### PR TITLE
Test that Positions library is memory safe

### PR DESCRIPTION
**Pull request by haydenshively**
forge vm added an `expectSafeMemory` [cheat](https://github.com/foundry-rs/foundry/pull/4491). Taking advantage of that for new tests.

### CODE REVIEW

File: core/test/libraries/Positions.t.sol [hunk 1]
```diff
@@ -19,6 +19,57 @@ contract PositionsTest is Test {
...
+            let ptr := mload(0x40)
+            mstore(ptr, dirt)
+            mstore(add(ptr, 0x20), dirt)
+            mstore(add(ptr, 0x40), dirt)
+            mstore(add(ptr, 0x60), dirt)
+            mstore(add(ptr, 0x80), dirt)
+            // ...could add more, doesn't really matter
+        }
+
+        int24[] memory extracted = extract(zipped);
```
Comment (haydenshively) [inline core/test/libraries/Positions.t.sol:59]: Goal is to make sure `extract` still works, even when the memory it's writing to is dirty (see this [big warning](https://docs.soliditylang.org/en/v0.8.19/internals/layout_in_memory.html))

File: core/test/libraries/Positions.t.sol [hunk 2]
```diff
@@ -19,6 +19,57 @@ contract PositionsTest is Test {
...
+        uint64 ptr;
+        assembly ("memory-safe") {
+            ptr := mload(0x40)
+        }
+
+        vm.expectSafeMemory(ptr, ptr + 32 * (1 + 2 * count));
+        extract(zipped);
+    }
+
+    function test_memoryExtractDirty(int24 xl, int24 xu, uint256 dirt) public {
```
Comment (aneesh1) [inline core/test/libraries/Positions.t.sol:44]: Good test 👍

File: core/test/libraries/Positions.t.sol [hunk 3]
```diff
@@ -19,6 +19,57 @@ contract PositionsTest is Test {
...
+        if (yl != yu) count += 1;
+        if (zl != zu) count += 1;
+        uint256 zipped = zip([xl, xu, yl, yu, zl, zu]);
+
+        uint64 ptr;
+        assembly ("memory-safe") {
+            ptr := mload(0x40)
+        }
+
+        vm.expectSafeMemory(ptr, ptr + 32 * (1 + 2 * count));
```
Comment (aneesh1) [inline core/test/libraries/Positions.t.sol:40]: Took some time to wrap my head around this, but this makes sense. Start at the free memory pointer and then move in 32 byte increments depending on how many elements there are in our compressed data format (which is just one number). 👍

**Review by aneesh1 — APPROVED** [review]
Looks good!

**Review by IanWoodard — APPROVED** [review]
While this does not prove the Positions library is memory safe, it is a good starting point and gives us more confidence than before.

**Pull request merged**